src/library_browser.js

```diff
@@ -1072,15 +1072,36 @@ mergeInto(LibraryManager.library, {
     }
 
     if (mode == 0 /*EM_TIMING_SETTIMEOUT*/) {
-      Browser.mainLoop.scheduler = function Browser_mainLoop_scheduler() {
+      Browser.mainLoop.scheduler = function Browser_mainLoop_scheduler_setTimeout() {
         setTimeout(Browser.mainLoop.runner, value); // doing this each time means that on exception, we stop
       };
       Browser.mainLoop.method = 'timeout';
     } else if (mode == 1 /*EM_TIMING_RAF*/) {
-      Browser.mainLoop.scheduler = function Browser_mainLoop_scheduler() {
+      Browser.mainLoop.scheduler = function Browser_mainLoop_scheduler_rAF() {
         Browser.requestAnimationFrame(Browser.mainLoop.runner);
       };
       Browser.mainLoop.method = 'rAF';
+    } else if (mode == 2 /*EM_TIMING_SETIMMEDIATE*/) {
+      if (!window.setImmediate) {
```

Comment:
i think you need `['setImmediate']` as closure almost certainly isn't aware of that name